lib/mix/lib/mix/compilers/test.ex

```diff
@@ -37,7 +37,11 @@ defmodule Mix.Compilers.Test do
         :noop
 
       [] ->
-        Mix.shell.error "Test patterns did not match any file: " <> Enum.join(test_patterns, ", ")
+        if test_patterns == [] do
+          Mix.shell.info "There are no tests to run"
+        else
+          Mix.shell.error "Test patterns did not match any file: " <> Enum.join(test_patterns, ", ")
+        end
```

Comment:
For consistency with the other clauses in this `case`, we could do this here:

```elixir
[] when test_patterns == [] ->
  Mix.shell.info "There are no tests to run"
  :noop

[] ->
  Mix.shell.error "Test patterns did not match any file: " <> Enum.join(test_patterns, ", ")
  :noop
```

Wdyt? :)